### PR TITLE
fix Error: Cannot find module on Windows OS

### PR DESCRIPTION
**Pull request by wizicer**
This patch can fix: https://github.com/o1-labs/snarkyjs/issues/896

### CODE REVIEW

File: js/node/node-backend.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import { isMainThread, parentPort, workerData, Worker } from 'worker_threads';
 import os from 'os';
 import wasm_ from '../../../_node_bindings/plonk_wasm.cjs';
-let filename = import.meta.url?.slice(7) ?? __filename;
+import { fileURLToPath } from 'url';
+let filename = fileURLToPath(import.meta.url);
```
Comment (mitschabaude) [inline js/node/node-backend.js:5]: nice! this won't work in the commonjs version though - can you cover the case where `import.meta.url` is undefined and use `__filename` in that case? thanks

**Review by mitschabaude — COMMENTED** [review]
thanks for contributing an important fix! just one small comment to address

File: js/node/node-backend.js (hunk 1 above)
Comment (wizicer) [reply js/node/node-backend.js]: Nice catch, I should considered this.

File: js/node/node-backend.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import { isMainThread, parentPort, workerData, Worker } from 'worker_threads';
 import os from 'os';
 import wasm_ from '../../../_node_bindings/plonk_wasm.cjs';
-let filename = import.meta.url?.slice(7) ?? __filename;
+import { fileURLToPath } from 'url';
+let filename = fileURLToPath(import.meta.url ?? __filename);
```
Comment (mitschabaude) [inline js/node/node-backend.js:5]:
```suggestion
let url = import.meta.url;
let filename = url !== undefined ? fileURLToPath(url) : __filename;
```
unfortunately it's slightly more complicated ^^

**Review by mitschabaude — APPROVED** [review]

**Pull request merged**